entry.go

```diff
@@ -67,33 +68,42 @@ func (entry *Entry) WithFields(fields Fields) *Entry {
 	for k, v := range fields {
 		data[k] = v
 	}
-	return &Entry{Logger: entry.Logger, Data: data}
+	return &Entry{Logger: entry.Logger, Data: data, Level: entry.Level}
 }
 
 // This function is not declared with a pointer value because otherwise
 // race conditions will occur when using multiple goroutines
 func (entry Entry) log(level Level, msg string) {
-	entry.Time = time.Now()
-	entry.Level = level
-	entry.Message = msg
+	var cxtEntry *Entry = &entry
+
+	// If the log level if different from the current stored log level
+	// Then we will create a copy of Entry to avoid messing with the original log level
```

Comment:
`entry Entry` is a non-pointer-receiver so it's already a copy and won't change the original one